modules/discoveryBidAdapter.js

```diff
@@ -0,0 +1,478 @@
+    const bids = getKv(serverResponse, 'body', 'seatbid', 0, 'bid');
+    const cur = getKv(serverResponse, 'body', 'cur');
+
+    const bidResponses = [];
+    for (let bid of bids) {
+      let impid = getKv(bid, 'impid');
+      if (itemMaps[impid]) {
+        let bidId = getKv(itemMaps[impid], 'req', 'bidId');
+        const mediaType = getKv(bid, 'w') ? 'banner' : 'native';
+        let bidResponse = {};
+        if (mediaType === 'native') {
+          const adm = getKv(bid, 'adm');
+          const admObj = JSON.parse(adm);
+          var native = {};
+          admObj.assets.forEach((asset) => {
+            if (asset.title) {
+              native.title = asset.title.text;
+            } else if (asset.data) {
+              native.data = asset.data.value;
+            } else if (asset.img) {
+              switch (asset.img.type) {
+                case 1:
+                  native.icon = {
+                    url: asset.img.url,
+                    width: asset.img.w,
+                    height: asset.img.h,
+                  };
+                  break;
+                default:
+                  native.image = {
+                    url: asset.img.url,
+                    width: asset.img.w,
+                    height: asset.img.h,
+                  };
+                  break;
+              }
+            }
+          });
+          if (admObj.link) {
+            if (admObj.link.url) {
+              native.clickUrl = admObj.link.url;
+            }
+          }
+          if (Array.isArray(admObj.eventtrackers)) {
+            native.impressionTrackers = [];
+            admObj.eventtrackers.forEach((tracker) => {
+              if (tracker.event !== 1) {
+                return;
+              }
+              switch (tracker.method) {
+                case 1:
+                  native.impressionTrackers.push(tracker.url);
+                  break;
+                // case 2:
+                //   native.javascriptTrackers = `<script src=\"${tracker.url}\"></script>`;
+                //   break;
+              }
+            });
+          }
+          if (admObj.purl) {
+            native.purl = admObj.purl;
+          }
+          bidResponse = {
+            requestId: bidId,
+            cpm: getKv(bid, 'price'),
+            creativeId: getKv(bid, 'cid'),
+            mediaType,
+            currency: cur,
+            netRevenue: true,
+            native,
+            ttl: TIME_TO_LIVE,
+            nurl: getKv(bid, 'nurl'),
+          };
+        } else {
+          bidResponse = {
+            requestId: bidId,
+            cpm: getKv(bid, 'price'),
+            width: getKv(bid, 'w'),
+            height: getKv(bid, 'h'),
+            creativeId: getKv(bid, 'cid'),
+            dealId: '',
+            mediaType,
+            currency: cur,
+            netRevenue: true,
+            ttl: TIME_TO_LIVE,
+            ad: getKv(bid, 'adm'),
+            nurl: getKv(bid, 'nurl'),
+          };
```

Comment:
I am not seeing meta.advertiserDomains in the above objects. This is also a requirement. Please review https://docs.prebid.org/dev-docs/bidder-adaptor.html#interpreting-the-response specifically the note regarding this property. Adding here as well:

```
Note: bid adapters must be coded to accept the ‘advertiserDomains’ parameter from their endpoint even if that endpoint doesn’t currently respond with that value. Prebid.org publishers have required that all bidders must eventually supply this value, so every bidder should be planning for it. There’s often a long lag time between making a PBJS adapter update and when most pubs upgrade to it, so we minimally require adapters to be ready for the day when the endpoint responds with adomain.
```

Reply:
ok, thanks for the careful guidance, I have changed it.